olp-cpp-sdk-core/include/olp/core/client/RetrySettings.h

```diff
@@ -67,9 +67,32 @@ struct CORE_API RetrySettings {
   int max_attempts = 3;
 
   /**
-   * @brief The connection timeout limit (in seconds).
+   * @brief Maximum time for request to complete (in seconds).
+   *
+   * The default value is 90.
+   *
+   * @note Connection or data transfer will be interrupted after specified
+   * period of time ignoring `connection_timeout` and `transfer_timeout` values.
+   */
+  int timeout = 90;
+
+  /**
+   * @brief Time allowed to wait for connection to establish (in milliseconds).
+   *
+   * The default value is 60 seconds.
+   *
+   * @note The value should be smaller than `timeout`.
+   */
+  std::chrono::milliseconds connection_timeout = std::chrono::seconds(60);
+
+  /**
+   * @brief Time allowed to wait without data transfer (in milliseconds).
+   *
+   * The default value is 30 seconds.
+   *
+   * @note The value should be smaller than `timeout`.
    */
-  int timeout = 60;
+  std::chrono::milliseconds transfer_timeout = std::chrono::seconds(30);
```

Comment:
Why connection is 60 but the transfer only 30? I would expect it vice versa, as usually most time spent downloading something and not just connecting

Reply:
It's a default values in NetworkSettings. Values form RetrySettings will be applied to NetworkSettings eventually.

Reply:
Usually we spend more time waiting for connection because of DNS name resolving, opening sockets, sending acknowlidge etc. When connection is established - data transfers over already existing 'channel', so it makes more sense to set this value lower.